### PR TITLE
Muon AlC interface crashes

### LINKED ISSUE #27221: Muon AlC interface crashes
### Expected behavior
If you enter the first set of data, tick `Auto` and press `Load` it should load the data

### Actual behavior
Hard crash

### Steps to reproduce the behavior
Set the default instrument to EMU, and turn on the data archive
`Interfaces`->`Muon`->`ALC`
Enter some valid data for the first run and tick the `Auto` box
Press `Load`
It crashes

### Platforms affected
Tested on windows but probably all

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
The ALCLatestFileFinder expected the instrument name to be case-consistent, e.g EMU, or emu. If it was supplied with a file with an upper case detector name, e.g EMU00066666 from //emu/data it would crash. The crash was due to querying an empty vector, which now has a return check, on the off chance the underlying directory changes (e.g wiped). The instrument check has also been changed so that it is now case-insensitive.

**To test:**
To replicate the original issue

    Set the default instrument to EMU, and turn off the data archive
    Add the //emu/data directory to user directories. 
    Interfaces->Muon->ALC
    Enter 66666
    Tick auto
    Press Load
    Should no longer crash (although this operation takes a long time as it loads a large amount of files in //emu/data )

Fixes #27221 . 
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
Can you add some tests for this behaviour.

**Comment by NickDraper** [comment]
You could create a separate issue to add the tests as long as that is prioritised after the release.

**Comment by StephenSmith25** [comment]
> Can you add some tests for this behaviour.

I've added two unittests, which should capture the behavior I've fixed.

**Review by Harrietbrown — APPROVED** [review]
Tests added to cover new functionality, all pass. good work, its good that we can get this in.

File: qt/scientific_interfaces/Muon/ALCLatestFileFinder.cpp [hunk 1]
```diff
@@ -25,6 +26,8 @@ std::string ALCLatestFileFinder::getMostRecentFile() const {
     // List all valid Nexus files in the directory
     Poco::Path path(m_firstRunFileName);
     std::vector<std::string> fileNames;
+    // if it found no valid files then return an error
+    // and retry disable the auto button
```
Comment (DanNixon) [inline qt/scientific_interfaces/Muon/ALCLatestFileFinder.cpp]: It's not important to fix this now, but could this have been worded a bit better.

The behavior described in the error condition is particular unclear (e.g. it mentions a button but this is not a view or presenter).

**Pull request merged**